ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/status/TotalBalances.java

```diff
@@ -22,77 +24,168 @@ public class TotalBalances {
+  }
+
+  UInt64 getRawCurrentEpochActiveValidators() {
+    return currentEpochActiveValidators;
+  }
+
+  UInt64 getRawPreviousEpochActiveValidators() {
+    return previousEpochActiveValidators;
+  }
+
+  UInt64 getRawCurrentEpochSourceAttesters() {
+    return currentEpochSourceAttesters;
+  }
+
+  UInt64 getRawCurrentEpochTargetAttesters() {
+    return currentEpochTargetAttesters;
+  }
+
+  UInt64 getRawCurrentEpochHeadAttesters() {
+    return currentEpochHeadAttesters;
+  }
+
+  UInt64 getRawPreviousEpochSourceAttesters() {
+    return previousEpochSourceAttesters;
+  }
+
+  UInt64 getRawPreviousEpochTargetAttesters() {
+    return previousEpochTargetAttesters;
+  }
+
+  UInt64 getRawPreviousEpochHeadAttesters() {
     return previousEpochHeadAttesters;
   }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    final TotalBalances that = (TotalBalances) o;
+    return Objects.equals(currentEpochActiveValidators, that.currentEpochActiveValidators)
+        && Objects.equals(previousEpochActiveValidators, that.previousEpochActiveValidators)
+        && Objects.equals(currentEpochSourceAttesters, that.currentEpochSourceAttesters)
+        && Objects.equals(currentEpochTargetAttesters, that.currentEpochTargetAttesters)
+        && Objects.equals(currentEpochHeadAttesters, that.currentEpochHeadAttesters)
+        && Objects.equals(previousEpochSourceAttesters, that.previousEpochSourceAttesters)
+        && Objects.equals(previousEpochTargetAttesters, that.previousEpochTargetAttesters)
+        && Objects.equals(previousEpochHeadAttesters, that.previousEpochHeadAttesters)
+        && Objects.equals(effectiveBalanceIncrement, that.effectiveBalanceIncrement);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        currentEpochActiveValidators,
+        previousEpochActiveValidators,
+        currentEpochSourceAttesters,
+        currentEpochTargetAttesters,
+        currentEpochHeadAttesters,
+        previousEpochSourceAttesters,
+        previousEpochTargetAttesters,
+        previousEpochHeadAttesters,
+        effectiveBalanceIncrement);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("currentEpochActiveValidators", currentEpochActiveValidators)
+        .add("previousEpochActiveValidators", previousEpochActiveValidators)
+        .add("currentEpochSourceAttesters", currentEpochSourceAttesters)
+        .add("currentEpochTargetAttesters", currentEpochTargetAttesters)
+        .add("currentEpochHeadAttesters", currentEpochHeadAttesters)
+        .add("previousEpochSourceAttesters", previousEpochSourceAttesters)
+        .add("previousEpochTargetAttesters", previousEpochTargetAttesters)
+        .add("previousEpochHeadAttesters", previousEpochHeadAttesters)
```

Comment:
did we intentionally leave out `effectiveBalanceIncrement` here?

Reply:
No.  It's not a particularly important value but may as well add it for good measure - might save some head scratching one day when we somehow make it different between two instances.